caas/kubernetes/provider/k8s.go

```diff
@@ -489,21 +490,22 @@ func (k *kubernetesClient) configureStorage(
 	if err != nil {
 		return errors.Trace(err)
 	}
-	for i, fs := range filesystems {
+	logger.Debugf("configuring pod filesystems: %+v", filesystems)
+	for _, fs := range filesystems {
 		var mountPath string
 		if fs.Attachment != nil {
 			mountPath = fs.Attachment.Path
 		}
 		if mountPath == "" {
 			mountPath = fmt.Sprintf("%s/fs/%s/%s", baseDir, appName, fs.Tag.Id())
 		}
-		pvcName := fmt.Sprintf("fsvolume-%d", i)
+		pvcNamePrefix := fs.Tag.String()
 		volStorageLabel := fmt.Sprintf("%s-unit-storage", appName)
 		params := volumeParams{
-			storageLabels: []string{volStorageLabel, k.namespace, "default"},
-			pvcName:       pvcName,
-			volumeSize:    fmt.Sprintf("%dMi", fs.Size),
-			labels:        map[string]string{labelApplication: appName},
+			storageLabels:       []string{volStorageLabel, k.namespace, "default"},
+			pvcName:             pvcNamePrefix,
+			requestedVolumeSize: fmt.Sprintf("%dMi", fs.Size),
```

Comment:
do we only allow `Mi`?

Reply:
The Juju state model stores the required sizes in MiB

Reply:
ic, thx